app/assets/javascripts/shared/ofn-loading.js.coffee

```diff
@@ -0,0 +1,15 @@
+angular.module("Loading", [])
+  .directive "loading", (Loading)->
+    # Triggers a screen-wide "loading" thing when Ajaxy stuff is happening
+    scope: {}
+    restrict: 'E'
+    templateUrl: 'loading.html'
+    controller: ($scope)->
+      $scope.Loading = Loading
+      $scope.show = ->
+        $scope.Loading.message?
+  .factory "Loading", ->
+    new class Loading
+      message: null
+      clear: =>
+        @message = null
```

Comment:
I think the convention is generally to name the file after the service/directive contained within, so there is no need for the `ofn-` prefix in the filename here.

Also, same deal here with the folder structure, directives in `directives`, factories in `services`. 🙏